vasyl_varyshchuk/Week_3/task_2/rest_api_requests.py

```diff
@@ -0,0 +1,95 @@
+"""
+This script contains functions for display, create, update and delete users from server with REST API
+"""
+import requests
+import json
+
+URL = 'http://127.0.0.1:5000/api/'
+headers = {'Content-type': 'application/json'}
+
+
+def get_list_of_users():
+    """This function returns list of users from a server"""
+    try:
+        response = requests.get(URL+'users/')
+        response.raise_for_status()
+        return response.json()['users']
+    except requests.RequestException as error:
+        print(error)
+
+
+def get_concrete_user(user_id):
+    """This function gets information about concrete user from server by user_id"""
+    try:
+        response = requests.get(URL+'users/{}/'.format(user_id))
+        response.raise_for_status()
+        return response.json()['user']
+    except requests.RequestException as error:
+        print (error)
+
+
+def add_new_user(new_user_info):
+    """
+    This function adds new user to the server passing a dictionary of information about user.
+    new_user_info parameter should be passed in the following format:
+        {'name': 'user_name', 'surname': 'user_surname', 'age': user_age}
+    """
+    new_user_json = json.dumps(new_user_info)
+    try:
+        response = requests.post(URL+'users/', data=new_user_json, headers=headers)
+        response.raise_for_status()
+        print('New user was successfully added')
+    except requests.RequestException as error:
+        print(error)
+
+
+def update_user_info(user_id, update_info):
+    """
+    This function updates information about current user.
+    update_info -  dictionary with new information about user
+    """
+    update_info_json = json.dumps(update_info)
+    user_url = URL + 'users/{}/'.format(user_id)
+    try:
+        get_user_response = requests.get(user_url)
+        put_response = requests.put(user_url, data=update_info_json, headers=headers)
+        put_response.raise_for_status()
+        if get_user_response.json()['user'] is not None:
```

Comment:
Do you really need this additional verification?

Reply:
In my case, I want to check whether a user with some id exists on the server before delete it or update information about this user. Without this verification, response.raise_for_status() returns 200 status code even user doesn't exist on the server. So, I get the same results when the user is present on the server and when a user is absent. Perhaps this can be verified in another way.